src/imap.rs

```diff
@@ -1625,14 +1625,17 @@ impl Imap {
         context
             .set_config_internal(Config::ConfiguredInboxFolder, Some("INBOX"))
             .await?;
+        for (config, name) in folder_configs {
+            context.set_config_internal(config, Some(&name)).await?;
+        }
         if let Some(mvbox_folder) = mvbox_folder {
             info!(context, "Setting MVBOX FOLDER TO {}", &mvbox_folder);
             context
                 .set_config_internal(Config::ConfiguredMvboxFolder, Some(mvbox_folder))
                 .await?;
-        }
-        for (config, name) in folder_configs {
-            context.set_config_internal(config, Some(&name)).await?;
+        } else if create_mvbox && context.config_exists(Config::MvboxMove).await? {
```

Comment:
This check doesn't look straightforward, better to remove `#[strum(props(default = "1"))]` for `MvboxMove` and make it dependent on the `IsChatmail` setting